src/libraries/System.Net.Security/src/System/Net/Security/NetSecurityTelemetry.cs

```diff
@@ -0,0 +1,288 @@
+                    break;
+            }
+
+            if (connectionOpen)
+            {
+                Interlocked.Increment(ref connectionsOpen);
+            }
+
+            if (handshakeDurationCounter != null)
+            {
+                handshakeDurationCounter.WriteMetric(stopwatch.GetElapsedTime().TotalMilliseconds);
+            }
+
+            if (IsEnabled(EventLevel.Informational, EventKeywords.None))
+            {
+                HandshakeStop(protocol.ToString());
+            }
+        }
+
+        [NonEvent]
+        public void ConnectionClosed(SslProtocols protocol)
+        {
+            long count = 0;
+
+            switch (protocol)
+            {
+                case SslProtocols.Tls:
+                    count = Interlocked.Decrement(ref _connectionsOpenTls10);
+                    break;
+
+                case SslProtocols.Tls11:
+                    count = Interlocked.Decrement(ref _connectionsOpenTls11);
+                    break;
+
+                case SslProtocols.Tls12:
+                    count = Interlocked.Decrement(ref _connectionsOpenTls12);
+                    break;
+
+                case SslProtocols.Tls13:
+                    count = Interlocked.Decrement(ref _connectionsOpenTls13);
+                    break;
+            }
+
+            Debug.Assert(count >= 0);
+        }
+
+
+        [NonEvent]
+        private unsafe void WriteEvent(int eventId, bool arg1, string? arg2)
+        {
+            if (IsEnabled())
+            {
+                arg2 ??= string.Empty;
+
+                fixed (char* arg2Ptr = arg2)
+                {
+                    const int NumEventDatas = 2;
+                    EventData* descrs = stackalloc EventData[NumEventDatas];
+
+                    descrs[0] = new EventData
+                    {
+                        DataPointer = (IntPtr)(&arg1),
+                        Size = sizeof(int) // EventSource defines bool as a 32-bit type
+                    };
+                    descrs[1] = new EventData
+                    {
+                        DataPointer = (IntPtr)(arg2Ptr),
+                        Size = (arg2.Length + 1) * sizeof(char)
+                    };
+
+                    WriteEventCore(eventId, NumEventDatas, descrs);
+                }
+            }
+        }
+
+        [NonEvent]
+        private unsafe void WriteEvent(int eventId, bool arg1, double arg2, string? arg3)
+        {
+            if (IsEnabled())
+            {
+                arg3 ??= string.Empty;
+
+                fixed (char* arg3Ptr = arg3)
+                {
+                    const int NumEventDatas = 3;
+                    EventData* descrs = stackalloc EventData[NumEventDatas];
+
+                    descrs[0] = new EventData
+                    {
+                        DataPointer = (IntPtr)(&arg1),
+                        Size = sizeof(int) // EventSource defines bool as a 32-bit type
```

Comment:
Same here - you will need to set `Reserved=0` explicitly for each of these.

Reply:
[Same comment as above](https://github.com/dotnet/runtime/pull/40108#discussion_r462687340)